Sources/MapboxCoreNavigation/EventDetails.swift

```diff
@@ -63,13 +76,26 @@ struct NavigationEventDetails: EventDetails {
                 state = UIApplication.shared.applicationState
             }
         }
-        
+
         return state
-    }()
-    let batteryLevel: Int = UIDevice.current.batteryLevel >= 0 ? Int(UIDevice.current.batteryLevel * 100) : -1
-    let batteryPluggedIn: Bool = [.charging, .full].contains(UIDevice.current.batteryState)
+    }
+    var batteryLevel: Int { UIDevice.current.batteryLevel >= 0 ? Int(UIDevice.current.batteryLevel * 100) : -1 }
+    var batteryPluggedIn: Bool { [.charging, .full].contains(UIDevice.current.batteryState) }
+    var device: String { UIDevice.current.machine }
+    var operatingSystem: String { "\(ProcessInfo.systemName) \(ProcessInfo.systemVersion)" }
+    var platform: String { ProcessInfo.systemName }
+    var sdkVersion: String {
+        guard let stringForShortVersion = Bundle.string(forMapboxCoreNavigationInfoDictionaryKey: "CFBundleShortVersionString") else {
+            preconditionFailure("CFBundleShortVersionString must be set in the Info.plist.")
+        }
+        return stringForShortVersion
+    }
+    var screenBrightness: Int { Int(UIScreen.main.brightness * 100) }
+    var volumeLevel: Int { Int(AVAudioSession.sharedInstance().outputVolume * 100) }
+}
+
+struct ActiveGuidanceEventDetails: NavigationEventDetails {
```

Comment:
This is the first time we’re making an explicit distinction between navigation and “active navigation”, but it does make some sense considering “passive navigation”. (So far, navigation would be assumed to be active unless otherwise specified.) I think “ActiveNavigation” would be preferable, because “guidance” is a term we normally restrict to guidance instructions across the Mapbox navigation stack, despite broader usage elsewhere in the industry.

Reply:
Thx, renamed the classes according to your comment.